docker/dev/Makefile

```diff
@@ -0,0 +1,69 @@
+BUILDDIR := .build
+APPDIR :=  panelapp
+SOURCEDIR := ../..
+
+# If the first argument is "command"...
+ifeq (command,$(firstword $(MAKECMDGOALS)))
+  # use the rest as arguments for "command"
+  COMMAND_ARGS := $(wordlist 2,$(words $(MAKECMDGOALS)),$(MAKECMDGOALS))
+  # ...and turn them into do-nothing targets
+  $(eval $(COMMAND_ARGS):;@:)
+endif
+
+OS := $(shell uname)
+ifeq ($(OS),Darwin)
+  ## LocalStack needs a different TMPDIR in OSX
+  TMPDIR := /private$(TMPDIR)
+endif
+
+clean: ## Remove build directory
+	rm -rf $(BUILDDIR)
+
+build: clean .copy_sources ## Build Docker Images
+	docker-compose build
+
+run: ## Run cluster
+	docker-compose up -d
+
+migrate: ## Create db schema, apply db migration (migrate command)
+	docker-compose exec web python manage.py migrate
+
+loaddata: ## Load genes files into db (loaddata command)
+	docker-compose exec web python manage.py loaddata /app/genes.json
+
+collectstatic: ## Deploy static files (collectstatic command)
+	docker-compose exec web python manage.py collectstatic --noinput
+
+stop: ## Stop cluster, withoud destroying
+	docker-compose stop
+
+start: ## Restart a stopped cluster
+	docker-compose start
+
+down: ## Destroy cluster
+	docker-compose down
+
+tests: ## Run tests
+	docker-compose exec web bash -c 'cd /app ; pytest'
```

Comment:
we've run with a problem with tests and any command that returns non-0 response - we are using `restart: on-failure` but when any test fails, it just keeps restarting non-stop

Reply:
But this is a `docker-compose exec`, not `run`, starting a secondary shell inside the running container. I'm quite sure Docker only monitors the main process for restarting.
But I will double-check.